src/database/firestore.ts

```diff
@@ -1,19 +1,22 @@
 import { app, auth } from 'firebase/app';
 import 'firebase/firestore';
 import 'firebase/auth';
-
 import { Observable } from 'rxjs/internal/Observable';
 
-export default class Firestore {
+import { Database } from './database';
+
+export default class Firestore implements Database {
     private app: app.App;
     private auth: auth.Auth;
+    private db: firebase.firestore.Firestore;
```

Comment:
can this type be simplified to firestore.Firestore?

Reply:
Nope, I get `Cannot find namespace 'firestore'`

Reply:
thats fine then